libs/maps/src/maps/CPointsMap.cpp

```diff
@@ -462,6 +424,23 @@ void CPointsMap::determineMatching2D(
 	local_y_max =
 		max(max(temp_nums[0], temp_nums[1]), max(temp_nums[2], temp_nums[3]));
 
+	for (size_t k = 0; k < nLocalPoints % 4; k++)
```

Comment:
It seems like it would be easier to handle the last 3 separately instead of padding with zeros.

Reply:
Thanks for the patch! Yes, it's clearer like that.